### ISSUE TITLE
Change to file lib/poller.php from release 1.2.18 to 1.2.19 introduced a bug that lead to problem reported in issue #4460 

### ISSUE THREAD

**Issue by chassepa** [Cacti/cacti#4490]
From release 1.2.18 to 1.2.19 the following change has been done to file lib/poller.php:

```
    554c577,581
    < 		if ($remainder && $limit != '') {
    ---
    > 		$rows = db_fetch_cell('SELECT COUNT(local_data_id)
    > 			FROM poller_output');
    > 
    > 		/* to much records in poller_output, process in chunks */
    > 		if ($rows && $remainder) {

```

I believe there is something wrong with the new test:

`if ($rows && $remainder) {`

which make the poller to enter into an infinite loop.

On our installation I changed it back to the old test:

`if ($remainder && $limit != '') {`

and it fixed the problem.

**Comment by netniV** [comment]
I believe that was already addressed in #4460

**Comment by chassepa** [comment]
@netniV You are right I should have written my comment directly in #4460. I close this issue.
